### PR TITLE
Airavata Desktop Client

### PR DESCRIPTION
**Pull request by ganning127**
Development updates have been sent out through mailing list.

### CODE REVIEW

File: airavata-local-agent/README.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# Airavata UI
+
```
Comment (DImuthuUpe) [inline airavata-local-agent/README.md:2]: Please provide a description of the local agent and what purpose it server. Also, please provide some screenshots

File: airavata-local-agent/electron-builder.yml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+appId: com.example.nextron
```
Comment (DImuthuUpe) [inline airavata-local-agent/electron-builder.yml:1]: Add Apache licence header for all files

File: airavata-local-agent/package.json [hunk 3]
```diff
@@ -0,0 +1,52 @@
+{
```
Comment (DImuthuUpe) [inline airavata-local-agent/package.json:1]: Please verify whether all the dependencies are apache licence compatible

File: airavata-local-agent/renderer/components/ExperimentsList.jsx [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+import relativeTime from "dayjs/plugin/relativeTime";
+dayjs.extend(relativeTime);
+
+
+const isValidStatus = (status) => {
+  let invalidStatus = ["CREATED"]; // TODO here
+  return !invalidStatus.includes(status);
+};
+
+const getExperimentApplication = (executionId) => {
```
Comment (DImuthuUpe) [inline airavata-local-agent/renderer/components/ExperimentsList.jsx:23]: Add a comment explaining why these values are hardcoded

File: airavata-local-agent/renderer/components/ExperimentsList.jsx [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+} from "@chakra-ui/react";
+import ExperimentModal from "./ExperimentModal";
+import { getColorScheme, getResourceFromId } from '../lib/utilityFuncs';
+import dayjs from "dayjs";
+import relativeTime from "dayjs/plugin/relativeTime";
+dayjs.extend(relativeTime);
+
+
+const isValidStatus = (status) => {
+  let invalidStatus = ["CREATED"]; // TODO here
```
Comment (DImuthuUpe) [inline airavata-local-agent/renderer/components/ExperimentsList.jsx:19]: What is the TODO here?

File: airavata-local-agent/renderer/components/JupyterLab.jsx [hunk 6]
```diff
@@ -0,0 +1,304 @@
...
+
+
+
+        <h1>Note: If you close this tab, your jupyter session will no longer save any changes.</h1>
+      </>
+    );
+  }
+}
+
+
```
Comment (DImuthuUpe) [inline airavata-local-agent/renderer/components/JupyterLab.jsx:163]: Do we need following commented code?

**Review by DImuthuUpe — CHANGES_REQUESTED** [review]
Looking great overall. I added some change requests. Specially, you need to add apache licence header for all the files

File: airavata-local-agent/renderer/components/JupyterLab.jsx (hunk 6 above)
Comment (ganning127) [reply airavata-local-agent/renderer/components/JupyterLab.jsx:163]: deleted

**Comment by ganning127** [comment]
depdencies (from package.json)
    "@ajna/pagination": "^1.4.19", // MIT
    "@chakra-ui/icons": "^2.1.1", // MIT
    "@chakra-ui/react": "^2.8.2", // MIT
    "@emotion/react": "^11.11.4", // MIT
    "@emotion/styled": "^11.11.5", // MIT
    "@novnc/novnc": "^1.3.0", // MIT
    "dayjs": "^1.11.11", // MIT
    "dockerode": "^4.0.2", // Apache-2.0
    "electron-is-dev": "^3.0.1", // MIT
    "electron-log": "^5.1.5", // MIT
    "electron-serve": "^1.3.0", // MIT
    "electron-store": "^8.2.0", // MIT
    "framer-motion": "^11.2.0", // MIT
    "protocol-registry": "^1.4.1", // MIT
    "react-icons": "^5.2.1", // MIT
    "react-loading": "^2.0.3", // MIT
    "react-syntax-highlighter": "^15.5.0", // MIT
    "react-vnc": "^1.0.0", // MIT
    "stream": "0.0.3", // MIT
    "tus-js-client": "^1.6.0", // MIT
    "update-electron-app": "^3.0.0", // MIT
    "usehooks-ts": "^3.1.0" // MIT


dev deps
    "@electron/fuses": "^1.8.0", // MIT
    "electron": "^29.3.0", // MIT
    "electron-builder": "^24.13.3", // MIT
    "next": "^13.5.6", // MIT
    "nextron": "^8.15.0", // MIT
    "react": "^18.2.0", // MIT
    "react-dom": "^18.2.0" // MIT

**Comment by DImuthuUpe** [comment]
Great. Thanks @ganning127

**Pull request merged**